src/openrct2/ride/Vehicle.cpp

```diff
@@ -9304,113 +9304,85 @@ int32_t Vehicle::UpdateTrackMotionMiniGolf(int32_t* outStation)
         }
     }
 
-    regs.eax = 0;
-    regs.dx = 0;
-    regs.ebx = 0;
+    int32_t sumAcceleration = 0;
+    int32_t numVehicles = 0;
     uint16_t totalMass = 0;
 
     for (Vehicle* vehicle = this;;)
     {
-        regs.ebx++;
-        regs.dx |= vehicle->update_flags;
+        numVehicles++;
         totalMass += vehicle->mass;
-        regs.eax += vehicle->acceleration;
-        regs.si = vehicle->next_vehicle_on_train;
-        if (static_cast<uint16_t>(regs.si) == SPRITE_INDEX_NULL)
+        sumAcceleration += vehicle->acceleration;
+        if (vehicle->next_vehicle_on_train == SPRITE_INDEX_NULL)
         {
             break;
         }
-        vehicle = GET_VEHICLE(static_cast<uint16_t>(regs.si));
+        vehicle = GET_VEHICLE(vehicle->next_vehicle_on_train);
     }
 
-    regs.eax /= regs.ebx;
-    regs.ecx = (regs.eax * 21) >> 9;
-    regs.eax = velocity >> 12;
-    regs.ecx -= regs.eax;
-    regs.ebx = velocity;
-    regs.edx = velocity >> 8;
-    regs.edx *= regs.edx;
-    if (regs.ebx < 0)
-    {
-        regs.edx = -regs.edx;
-    }
-    regs.edx >>= 4;
-    regs.eax = regs.edx / totalMass;
-    regs.ecx -= regs.eax;
+    int32_t newAcceleration = (sumAcceleration / numVehicles * 21) >> 9;
+    newAcceleration -= velocity >> 12;
+    newAcceleration -= (((velocity >> 8) * abs(velocity >> 8)) >> 4) / totalMass;
```

Comment:
(vehicle >> 8) was squared in the old code.

Reply:
I thought so too at first glance, but the actuall math is `var * abs(var)`

This was done like
```
newvar = var * var
if (var < 0)
    newvar = -newvar
```

Reply:
I can create an extra variable to keep it the old way, but I'd need your help coming up with a name for it 😃